registration/include/pcl/registration/icp.h

```diff
@@ -93,46 +93,49 @@ namespace pcl
       using PointIndicesConstPtr = PointIndices::ConstPtr;
 
-      using Ptr = shared_ptr<IterativeClosestPoint<PointSource, PointTarget, Scalar> >;
-      using ConstPtr = shared_ptr<const IterativeClosestPoint<PointSource, PointTarget, Scalar> >;
-
-      using Registration<PointSource, PointTarget, Scalar>::reg_name_;
-      using Registration<PointSource, PointTarget, Scalar>::getClassName;
-      using Registration<PointSource, PointTarget, Scalar>::input_;
-      using Registration<PointSource, PointTarget, Scalar>::indices_;
-      using Registration<PointSource, PointTarget, Scalar>::target_;
-      using Registration<PointSource, PointTarget, Scalar>::nr_iterations_;
-      using Registration<PointSource, PointTarget, Scalar>::max_iterations_;
-      using Registration<PointSource, PointTarget, Scalar>::previous_transformation_;
-      using Registration<PointSource, PointTarget, Scalar>::final_transformation_;
-      using Registration<PointSource, PointTarget, Scalar>::transformation_;
-      using Registration<PointSource, PointTarget, Scalar>::transformation_epsilon_;
-      using Registration<PointSource, PointTarget, Scalar>::transformation_rotation_epsilon_;
-      using Registration<PointSource, PointTarget, Scalar>::converged_;
-      using Registration<PointSource, PointTarget, Scalar>::corr_dist_threshold_;
-      using Registration<PointSource, PointTarget, Scalar>::inlier_threshold_;
-      using Registration<PointSource, PointTarget, Scalar>::min_number_correspondences_;
-      using Registration<PointSource, PointTarget, Scalar>::update_visualizer_;
-      using Registration<PointSource, PointTarget, Scalar>::euclidean_fitness_epsilon_;
-      using Registration<PointSource, PointTarget, Scalar>::correspondences_;
-      using Registration<PointSource, PointTarget, Scalar>::transformation_estimation_;
-      using Registration<PointSource, PointTarget, Scalar>::correspondence_estimation_;
-      using Registration<PointSource, PointTarget, Scalar>::correspondence_rejectors_;
+      using Ptr = shared_ptr<Self>;
+      using ConstPtr = shared_ptr<const Self>;
+
+      using Base::reg_name_;
+      using Base::getClassName;
+      using Base::input_;
+      using Base::indices_;
+      using Base::target_;
+      using Base::nr_iterations_;
+      using Base::max_iterations_;
+      using Base::previous_transformation_;
+      using Base::final_transformation_;
+      using Base::transformation_;
+      using Base::transformation_epsilon_;
+      using Base::transformation_rotation_epsilon_;
+      using Base::converged_;
+      using Base::corr_dist_threshold_;
+      using Base::inlier_threshold_;
+      using Base::min_number_correspondences_;
+      using Base::update_visualizer_;
+      using Base::euclidean_fitness_epsilon_;
+      using Base::correspondences_;
+      using Base::transformation_estimation_;
+      using Base::correspondence_estimation_;
+      using Base::correspondence_rejectors_;
```

Comment:
This is horror! We should move them into protected access instead of public access (in another PR)